security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -351,16 +575,186 @@ private static LimitViolationKey toKey(LimitViolation violation) {
+                                                      LimitViolationFilter limitViolationFilter, boolean filterPreContingencyViolations) {
+        printPostContingencyViolations(result, network, writer, formatterFactory, TableFormatterConfig.load(), limitViolationFilter, filterPreContingencyViolations);
+    }
+
+
+    public static void printPostContingencyViolations(SecurityAnalysisResult result, Network network, Writer writer, TableFormatterFactory formatterFactory,
+            TableFormatterConfig formatterConfig, LimitViolationFilter limitViolationFilter, boolean filterPreContingencyViolations) {
+        Objects.requireNonNull(result);
+        Objects.requireNonNull(writer);
+        Objects.requireNonNull(formatterFactory);
+        if (!result.getPostContingencyResults().isEmpty()) {
+            Set<LimitViolationKey> preContingencyViolations = filterPreContingencyViolations
+                    ? result.getPreContingencyResult().getLimitViolations()
+                            .stream()
+                            .map(Security::toKey)
+                            .collect(Collectors.toSet())
+                    : Collections.emptySet();
+
+            NumberFormat numberFormat = getFormatter(formatterConfig.getLocale(), 4, 4);
+            NumberFormat percentageFormat = getFormatter(formatterConfig.getLocale(), 2, 2);
+
+            int sumFilter = result.getPostContingencyResults()
+                .stream()
+                .sorted(Comparator.comparing(o2 -> o2.getContingency().getId()))
+                .mapToInt(postContingencyResult -> {
+                    // configured filtering
+                    List<LimitViolation> filteredLimitViolations = limitViolationFilter != null
+                            ? limitViolationFilter.apply(postContingencyResult.getLimitViolationsResult().getLimitViolations(), network)
+                            : postContingencyResult.getLimitViolationsResult().getLimitViolations();
+
+                    // pre-contingency violations filtering
+                    List<LimitViolation> filteredLimitViolations2 = filteredLimitViolations.stream()
+                            .filter(violation -> preContingencyViolations.isEmpty() || !preContingencyViolations.contains(toKey(violation)))
+                            .collect(Collectors.toList());
+
+                    return filteredLimitViolations2.size();
+                }
+               ).sum();
+
+            try (TableFormatter formatter = formatterFactory.create(writer,
+                "Post-contingency limit violations",
+                formatterConfig,
+                new Column(CONTINGENCY),
+                new Column(STATUS),
+                new Column(ACTION),
+                new Column(COUNTRY),
+                new Column(BASE_VOLTAGE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT),
```

Comment:
country and base voltage columns are related to the voltage level. Move these columns after the "END" column

Reply:
done